oauth2/authorize.php

```diff
@@ -36,6 +37,14 @@
     exit;
 }
 
+// ensure 1) site from gbl and globals are the same and 2) ensure the site exists on filesystem
+if (empty($gbl::$SITE) || empty($_SESSION['site_id']) || ($gbl::$SITE != $_SESSION['site_id']) || !file_exists($GLOBALS['OE_SITES_BASE'] . '/' . $_SESSION['site_id'])) {
+    // error collecting site
+    error_log("OpenEMR error - oauth2 error since unable to properly collect site, so forced exit");
+    http_response_code(500);
```

Comment:
Keeping in mind we are really an interface class we need to inform client that's it their fault for not sending a valid request so  http_response_code(400).
Also in order to maintain compliance, eventually we need to go through all our responses using http_response_code() and build a psr error response that includes status, error code, error description and a hint.
This is easier by throwing an exception and emitting response from there.

Reply:
Happy to start chipping away at this. Do we have any example in codebase now that I can mimick?

Reply:
https://github.com/openemr/openemr/blob/3fca96b5e92f730c4b4b82c7270a945b17f6a8e5/src/RestControllers/AuthorizationController.php#L179

Except for responses outside controller(rest config) then i'd extend Exception with a new function in rest config and build and emit response there.